apps/www/_blog/2024-08-13-supabase-realtime-broadcast-and-presence-authorization.mdx

```diff
@@ -0,0 +1,197 @@
+to authenticated
+using ( true );
+
+create policy "authenticated user write to all"
+on "realtime"."messages"
+as permissive for insert
+to authenticated
+with check ( true );
+```
+
+We also have a new database function called`realtime.topic()` which enables you to write Policies that take into consideration the name of the Channel trying to run the RLS Policy.
+
+```sql
+create policy "authenticated users can only read from 'locked' topic"
+on "realtime"."messages"
+as permissive for select
+to authenticated
+using ( realtime.topic() = 'locked' );
+```
+
+### Enabling Authorization on Channel subscription
+
+Now that you have your RLS policies set, you will need to define on your Channel subscription that you intend to use a private channel. To achieve it we've introduced a new configuration parameter `private` to signal to Realtime servers that you want to check authorization on the channel.
+
+If you try to subscribe with an unauthorized user you will get a new error message informing the user that they do not have permission to access the topic.
+
+```tsx
+// With anon user
+supabase.realtime
+  .channel('locked', { config: { private: true } })
+  .subscribe((status: string, err: any) => {
+    if (status === 'CHANNEL_ERROR') {
+      console.error(err.message)
+    }
+  })
+
+// Outputs the following error:
+// "You do not have permissions to read from this Topic"
+```
+
+But if you connect with an authorized user you will be able to listen to all messages from the “private” topic
+
+```tsx
+// With an authenticated user
+supabase.realtime.setAuth(token)
+
+supabase.realtime
+  .channel('locked', { config: { private: true } })
+  .subscribe((status: string, err: any) => {
+    if (status === 'CHANNEL_ERROR') {
+      console.error(err.message)
+    }
+  })
+```
+
+### Advanced examples
+
+You can find a more complex example in the [Supabase demo folder](https://github.com/supabase/supabase/tree/master/examples/realtime/nextjs-authorization-demo) where we are using this feature to build chat rooms with restricted access.
+
+## How does it work?
+
+We decided on an approach that keeps your database and RLS policies at the heart of the new authorization strategy.
+
+### Database as a source of security
+
+To achieve Realtime authorization, we looked into our current solutions, namely how [Storage](https://supabase.com/docs/guides/storage/security/access-control) handles Access Control. Due to the nature of Realtime, our primitives are different as we have no assets stored in the database. So how did we achieve it?
+
+We let you define that the channel itself is private! On Channel subscription you are able to inform Realtime to use a private channel and we will do the required checks.
+
+The checks are done by running SELECT and INSERT queries on the new `realtime.messages` table which are then rolled backed so nothing is persisted. Then, based on the query result, we can determine the policies the user has for a given extension.
+
```

Comment:
@filipecabaco can we add an example of the policy we generate on the Elixir side? 

```
The policy generated on the Realtime server looks like: 

[elixir policy here]
```

And maybe link to where we do this in the repo?